indexing-service/src/main/java/io/druid/indexing/overlord/supervisor/SupervisorManager.java

```diff
@@ -74,11 +74,13 @@ public boolean createAndStartSupervisor(SupervisorSpec spec)
     return createAndStartSupervisorInternal(spec, true);
   }
 
-  public void stopAndRemoveSupervisor(String id)
+  public void stopAndRemoveSupervisor(String id, boolean writeTombstone)
   {
     Pair<Supervisor, SupervisorSpec> pair = supervisors.get(id);
     if (pair != null) {
-      metadataSupervisorManager.insert(id, new NoopSupervisorSpec()); // where NoopSupervisorSpec is a tombstone
+      if (writeTombstone) {
```

Comment:
Skip writing the tombstone if we're doing a schema migration to avoid cluttering the audit trail.